### DISCUSSION TITLE
Large Dataset vs Small Dataset

### DISCUSSION THREAD

**Discussion by blizaga** [explosion/spaCy#13007] [Help: Best practices]
I would like to ask for texcat model with tokenizer transformer, is the best practice with a large dataset or with a dataset that is not too much to use when training the model?

Because when I did a benchmark with spaCy's default commands, for a large dataset the score was not as big as when using a smaller dataset.

**Comment by shadeMe** [comment]
When training models, it's preferable to use as much training data as possible but one must also take diversity into account. A large dataset that only has examples of a small subset of labels/categories will perform relatively poorly over a smaller dataset with a more diverse set of examples. The important point of consideration when it comes to training any model is to ensure that the training data is representative/diverse enough to allow the model to generalize over a wide range on inputs.

**Reply by blizaga** [reply]
As for the score that appears at the end of the train on the spacy, can it be trusted immediately?

**Reply by shadeMe** [reply]
The score corresponds to the prediction performance of the model on the validation/evaluation data. If that data is representative of the inputs the model will see during inference, it should be a reliable indicator.

**Reply by blizaga** [reply]
I divided the dataset into 80% train data, 10% validation data, and 10% test data. is that correct?

**Reply by blizaga** [reply]
Of course, I also did a bencmark evaluation according to the spacy command line interface.
